src/java/com/wolfssl/provider/jsse/WolfSSLContext.java

```diff
@@ -125,7 +125,7 @@ private void createCtx() throws WolfSSLException {
         }
 
         /* auto-populate enabled protocols with supported ones */
-        params.setProtocols(WolfSSL.getProtocols());
+        params.setProtocols(this.getProtocolsMask(ctxAttr.noOptions));
```

Comment:
We may want to do this check before creating the "new com.wolfssl.WolfSSLContext(method)" above and throw an exception if "method" is not in the the allowed protocol list?

Reply:
In native wolfSSL, conflict like following case is accepted, for example.
    ctx = CTX_new(TLS1_3_Method);
    SetOptions(ctx, NO_TLS1_3);
Same may be applied to JSSE layer?

Reply:
It being counter intuitive was what I was thinking, yes.  In native wolfSSL, the user knows that they are explicitly setting the options on the ctx.  In wolfJSSE, this all happens at this point when the user creates a SSLContext, ex:

```
SSLContext ctx = SSLContext.getInstance("TLSv1.3");
```

My thought was that if they try to make a context with a specific protocol version (example: TLSv1.3), they should get some kind of error if the context was not able to be created because of the custom user callback prohibiting that version.  Otherwise, their context would "downgrade" to a lower or different protocol version without them knowing.

Thanks for making the adjustment!